test/support/utils.h

```diff
@@ -777,6 +777,37 @@ struct UserBinaryPredicate
     }
 };
 
+template <typename T>
+struct MatrixPoint
```

Comment:
Instead of defining a new class here, can we use something like `sycl::vec<T, 2>` or even `std::pair<T>`? I believe it'll effectively have the same level of test coverage but with less code.

Reply:
I don't think `sycl::vec` will work here since it will also go through host policy testing. Also, since `std::pair` doesn't have an `operator+` we would need to remove the testing that relies on the default binary operator.

Reply:
Ah, okay. Then I think it's fine to leave it as is in that case.